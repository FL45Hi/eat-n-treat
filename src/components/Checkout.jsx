import React from "react";

function Checkout() {
  const restID = localStorage.getItem("rID");
  const cartItems = JSON.parse(localStorage.getItem("cart"));
  console.log(cartItems);
  return (
    <div className="container" style={{ marginTop: "4rem" }}>
      <h1>Order Details</h1>
      <table>
        <thead>
          <tr>
            <th>Item Name</th>
            <th>Qty</th>
            <th>Price</th>
            <th>Q x P</th>
          </tr>
        </thead>
        <tbody>
          {cartItems.map((item) => (
            <tr>
              <td>{item.menuName}</td>
              <td>{item.qty}</td>
              <td>{item.menuPrice}</td>
              <th>{item.qty * item.menuPrice}</th>
            </tr>
          ))}
        </tbody>
      </table>
      <h4>
        Total :{" "}
        {cartItems.reduce(
          (tot, current) => tot + current.menuPrice * current.qty,
          0
        )}
      </h4>
      <button>Place order</button>
    </div>
  );
}

export default Checkout;
