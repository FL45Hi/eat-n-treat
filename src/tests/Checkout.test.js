import React from 'react';
import renderer from 'react-test-renderer';

import Checkout from '../components/Checkout';

it('Testing Checkout.jsx snapshot', () => {
    if(localStorage.getItem("cart") === null || JSON.parse(localStorage.getItem("cart")).length === 0) {
        localStorage.setItem("cart", JSON.stringify([
            {
                "menuName": "Chai",
                "menuPrice": 20,
                "_id": "614d7b246f174a678b9ba230",
                "qty": 4
            },
            {
                "menuPrice": 40,
                "menuName": "Cold Coffe",
                "_id": "614ded911ce7a88f83bbc55b",
                "qty": 1
            }
        ]))
    }

    console.log(JSON.parse(localStorage.getItem("cart")))

    const tree = renderer.create(<Checkout />).toJSON();
    expect(tree).toMatchSnapshot()
})