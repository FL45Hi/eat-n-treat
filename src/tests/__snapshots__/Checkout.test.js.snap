// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Checkout.jsx snapshot 1`] = `
<div
  className="container"
  style={
    Object {
      "marginTop": "4rem",
    }
  }
>
  <h1>
    Order Details
  </h1>
  <table>
    <thead>
      <tr>
        <th>
          Item Name
        </th>
        <th>
          Qty
        </th>
        <th>
          Price
        </th>
        <th>
          Q x P
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Chai
        </td>
        <td>
          4
        </td>
        <td>
          20
        </td>
        <th>
          80
        </th>
      </tr>
      <tr>
        <td>
          Cold Coffe
        </td>
        <td>
          1
        </td>
        <td>
          40
        </td>
        <th>
          40
        </th>
      </tr>
    </tbody>
  </table>
  <h4>
    Total :
     
    120
  </h4>
  <button>
    Place order
  </button>
</div>
`;
